src/controller_examples/simple_controller/spec/custom_resource.rs

```diff
@@ -133,15 +141,56 @@ impl ResourceView for CustomResourceView {
         }
     }
 
-    open spec fn from_dynamic_object(obj: DynamicObjectView) -> CustomResourceView {
-        CustomResourceView {
-            metadata: obj.metadata,
-            spec: CustomResourceSpecView{
-                content: obj.data.get_Object_0()[spec_field()].get_Object_0()[spec_content_field()].get_String_0(),
-            },
-            status: if obj.data.get_Object_0()[status_field()].is_Null() {Option::None} else {Option::Some(CustomResourceStatusView{
-                echoed_content: obj.data.get_Object_0()[status_field()].get_Object_0()[status_echoed_content_field()].get_String_0(),
-            })},
+    open spec fn from_dynamic_object(obj: DynamicObjectView) -> Result<CustomResourceView, ParseDynamicObjectError> {
```

Comment:
You might want to avoid super long nested if-else because it hurts readability.
Could you refactor the code in the following format:
```
if ... {
  Error1
} else if ... {
  Error2
} else {
  Ok(...)
}
```
Note that you can "prefetch" the content in the enum like `let x = y.get_Some_0()` before checking the variant like `y.is_Some()` -- it still makes sense as long as you use `x` in a branch where `y.is_Some()` is true.

Reply:
No problem